### PR TITLE
Allow configuration of confirmation prompt comparison via `StringComparer`

### PR DESCRIPTION
**Pull request by MartinZikmund**
Based on suggestion from @0xced https://github.com/spectreconsole/spectre.console/pull/1151#issuecomment-1426768481 👍

### CODE REVIEW

File: src/Spectre.Console/Prompts/ConfirmationPrompt.cs [hunk 1]
```diff
@@ -40,10 +40,13 @@ public sealed class ConfirmationPrompt : IPrompt<bool>
...
-    /// Gets or sets a value indicating whether the confirmation
-    /// should use case insensitive matching.
+    /// Gets or sets the string comparer to use when comparing user input
+    /// against Yes/No choices.
     /// </summary>
-    public bool CaseInsensitive { get; set; } = true;
+    /// <remarks>
+    /// Defaults to <see cref="StringComparer.CurrentCultureIgnoreCase"/>.
+    /// </remarks>
+    public StringComparer ChoiceComparer { get; set; } = StringComparer.CurrentCultureIgnoreCase;
```
Comment (patriksvensson) [inline src/Spectre.Console/Prompts/ConfirmationPrompt.cs:49]: Perhaps we can just call it `Comparer`.

File: src/Spectre.Console/Prompts/ConfirmationPrompt.cs [hunk 2]
```diff
@@ -75,6 +77,6 @@ public async Task<bool> ShowAsync(IAnsiConsole console, CancellationToken cancel
 
         var result = await prompt.ShowAsync(console, cancellationToken).ConfigureAwait(false);
 
-        return Yes.ToString().Equals(result.ToString(), CaseInsensitive ? StringComparison.CurrentCultureIgnoreCase : StringComparison.CurrentCulture);
+        return ChoiceComparer.Compare(Yes.ToString(), result.ToString()) == 0;
```
Comment (patriksvensson) [inline src/Spectre.Console/Prompts/ConfirmationPrompt.cs:80]: The comparer might be null here, so we need to fall back on the case-insensitive one if it is.

**Review by patriksvensson — CHANGES_REQUESTED** [review]

**Comment by MartinZikmund** [comment]
@patriksvensson Sorry for the delay, made adjustments 👍

**Comment by patriksvensson** [comment]
@MartinZikmund No worries. No SLA for open source contributions 😉

**Review by patriksvensson — APPROVED** [review]

**Comment by patriksvensson** [comment]
Merged! Thank you for your contribution. Much appreciated! 👍

**Pull request merged**